### PR TITLE
feat: initial OAuth support

### PR DESCRIPTION
**Pull request by MartinKolarik**
Adds support for:
 - authorization_code & refresh_token,
 - token revocation (supports tokens manually created via the dashboard too),
 - token introspection (supports token manually created via the dashboard too).

In the end, I extracted this to be a standalone app as there is little code overlap with the other projects, and it'll be easier to manage this way. The goal is to get this live ASAP to support https://github.com/jsdelivr/globalping-cli/issues/108 and, a bit later, open the server to other apps.

### CODE REVIEW

File: src/oauth/model.ts [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+			return code;
+		}
+
+		const newKey = this.getApprovedAuthorizationCodeRedisKey(code.authorizationCode);
+		await this.redis.rename(oldKey, newKey);
+
+		return code;
+	}
+
+	async saveAuthorizationCode (code: AuthorizationCodeToSave, client: ClientWithCredentials, user: User): Promise<AuthorizationCode | null> {
```
Comment (alexey-yarmosh) [inline src/oauth/model.ts:164]: That function may return `null`? Same in `saveToken`.

File: src/oauth/model.ts [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+		]);
+
+		return {
+			...code,
+			client,
+			user,
+		};
+	}
+
+	async revokeAuthorizationCode (code: AuthorizationCode): Promise<boolean> {
```
Comment (alexey-yarmosh) [inline src/oauth/model.ts:197]: Shouldn't that method revoke pending codes as well?

Comment (MartinKolarik) [reply src/oauth/model.ts:197]: There's no flow where that would be needed. Revocation happens only when you use the code and you can't use the one that's pending. Also regardless of approval the codes expire themselves after five minutes.

File: src/oauth/model.ts (hunk 1 above)
Comment (MartinKolarik) [reply src/oauth/model.ts:164]: Not in our case, the signature was based on the parent interface which allows it as an alternative to throwing errors.

**Review by alexey-yarmosh — APPROVED** [review]

**Pull request merged**